### DISCUSSION TITLE
Defining supported versions of MongoDB, OTP and Elixir

### DISCUSSION THREAD

**Discussion by joeapearson** [ankhers/mongodb#372] [Ideas]
I propose we drop support for versions of MongoDB that MongoDB themselves no longer support:

https://docs.mongodb.com/manual/release-notes/

TL;DR; Mongo 4.0 and above will be "officially supported".  Older versions may still work but are "at your own risk".

I suggest we take a soft deprecation approach:

1. Immediately update the docs to indicate this change but make no changes to the code
2. Gradually as we go along remove any Mongo < 4.0 specific code as needed

Doing so will help us minimize maintenance overhead and keep this library alive.  Users can still use previous published versions of the package if they really want to.

Let me know what you think!

**Comment by scottmessinger** [comment]
I completely agree.

**Comment by joeapearson** [comment]
@ankhers has pointed out in #370 that we also need to similarly define supported versions of Elixir + OTP.  I suggest a similar approach of soft deprecations where possible but only providing official support for those versions of Elixir and OTP that are themselves officially supported.

It's not that I don't care about users on older versions but I am of the opinion that we should try hard to keep things as manageable as possible.
